packages/hash/frontend/src/components/hooks/use-get-workspace-shortname-by-entity-uuid.ts

```diff
@@ -0,0 +1,75 @@
+import { extractBaseUri } from "@blockprotocol/type-system";
+import { systemUserShortname } from "@hashintel/hash-shared/environment";
+import { types } from "@hashintel/hash-shared/ontology-types";
+import { entityIdFromOwnedByIdAndEntityUuid } from "@hashintel/hash-subgraph";
+import { getRootsAsEntities } from "@hashintel/hash-subgraph/src/stdlib/element/entity";
+import { useEffect, useMemo, useState } from "react";
+import { useBlockProtocolGetEntity } from "./blockProtocolFunctions/knowledge/useBlockProtocolGetEntity";
+import { useGetWorkspaceByShortname } from "./use-get-workspace-by-shortname";
+
+export const useGetWorkspaceShortnameByEntityUuid = (params: {
+  entityUuid: string;
+}): { workspaceShortname?: string; loading: boolean } => {
+  const { entityUuid } = params;
+
+  const [workspaceShortname, setWorkspaceShortname] = useState<string>();
+  const [loading, setLoading] = useState<boolean>(false);
+
+  const { getEntity } = useBlockProtocolGetEntity();
+
+  const {
+    workspace: systemUserWorkspace,
+    loading: loadingSystemUserWorkspace,
+  } = useGetWorkspaceByShortname(systemUserShortname);
+
+  const systemUserOwnedById = useMemo(
+    () =>
+      systemUserWorkspace
+        ? systemUserWorkspace.kind === "user"
+          ? systemUserWorkspace.userAccountId
+          : systemUserWorkspace.orgAccountId
```

Comment:
I think the reason I avoided this originally was because I didn't want to introduce usages of `accountId` in the codebase at a time when it was used in way too many places. This has been resolved to a large extent thanks to the introduction of `ownedByid` in the FE, so am happy to make this change now: https://github.com/hashintel/hash/pull/1650/commits/936a4734986f4d4cdfdc74feb321c997728a2e8c

Reply:
Makes sense. Especially with type branding (which I'll make sure to update for this PR after it's merged), it will be very clear what the types are :)